src/commands/azureServiceOperators/helpers/azureservicehelper.ts

```diff
@@ -0,0 +1,334 @@
+    try {
+        const kubectl = await k8s.extension.kubectl.v1;
+        let runResult;
+        if (kubectl.available) {
+            const asoOlmYamlFile = "https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.17.0/olm.yaml";
+
+            runResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+                clusterKubeConfig, "YAML",
+                (f) => kubectl.api.invokeCommand(`create -f ${asoOlmYamlFile} --kubeconfig="${f}"`));
+        }
+        return runResult;
+
+    } catch (e) {
+        vscode.window.showErrorMessage(`ASO OLM Deployment resource had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export async function runOLMASOYaml(clusterKubeConfig: string): Promise<k8s.KubectlV1.ShellResult | undefined> {
+    try {
+        const kubectl = await k8s.extension.kubectl.v1;
+        let runResult;
+        if (kubectl.available) {
+            const asoYamlFile = "https://operatorhub.io/install/azure-service-operator.yaml";
+
+            runResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+                clusterKubeConfig, "YAML",
+                (f) => kubectl.api.invokeCommand(`create -f ${asoYamlFile} --kubeconfig="${f}"`));
+        }
+        return runResult;
+
+    } catch (e) {
+        vscode.window.showErrorMessage(`ASO Deployment resoruce had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export async function runASOIssuerCertYAML(clusterKubeConfig: string): Promise<k8s.KubectlV1.ShellResult | undefined> {
+    try {
+        const extensionPath = getExtensionPath();
+        const tempFile = tmp.fileSync({ prefix: "aso-issuer", postfix: `.yaml` });
+
+        const yamlPathOnDisk = vscode.Uri.file(path.join(extensionPath!, 'resources', 'yaml', 'issuerandcertmanager.yaml'));
+
+        const issuerandcertmanager = fs.readFileSync(yamlPathOnDisk.fsPath, 'utf8');
+        fs.writeFileSync(tempFile.name, issuerandcertmanager);
+
+        const kubectl = await k8s.extension.kubectl.v1;
+        let runCommandResult;
+        if (kubectl.available) {
+            runCommandResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+                clusterKubeConfig, "YAML",
+                (f) => kubectl.api.invokeCommand(`apply -f ${tempFile.name} --kubeconfig="${f}"`));
+        }
+
+        return runCommandResult;
+    } catch (e) {
+        vscode.window.showErrorMessage(`ASO Issuer and Cert Deployment file had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export async function getAzureServicePrincipal(
+    target: AksClusterTreeItem
+): Promise<OperatorSettings | undefined> {
+    // Note: Will be removed soon.
+    // So what is happenign here?
+    // There is no way I could see we can generate ServicePricipal credential for the user.
```

Comment:
💡 Old comment indeed: essentially the `servicepriciple` here are the few gotchas: 

We decided to get user input for `AppID and Password`  

* This `sp create` is not actually create all the time, if the `service principle is existing` it does `update` and gives back new `password` or `clientsecret` everytime. 
* From Portal, when creating `service principle` the password generated comes with warning that `user should save this passwrod somewhere because this is the only time they are seeing this password`.
* Also, found out that for AZ CLI `2.13.1 and below` this create command is not working.

Hence it made more sense that instead of generating new password for the user all the time for the user, `User provide us` their AppID and their saved credentials.

Reply:
💡 📓  Note:

Another idea: (yet to reachout to AAD and some ASO folks to check on this)
* What I will again try and reachout is that if the AAD folks have any `sdk` and we can use it internally at vscode level to generate the unique `ServicePrinciple` and then we can do more organic, no user input required scenario if that is feasiable.